components/prism-wgsl.js

```diff
@@ -0,0 +1,67 @@
+Prism.languages.wgsl = {
+    'comment': {
+        pattern: /\/\/.*|\/\*[\s\S]*?(?:\*\/|$)/,
+        greedy: true,
+    },
+    'attributes': {
+        pattern: /(@)(?:align|binding|builtin|const|group|id|interpolate|invariant|location|size|workgroup_size|vertex|fragment|compute)/i,
+        lookbehind: true,
+        alias: 'attr-name',
+    },
+    'functions': {
+        pattern: /\b(fn\s+)[_a-zA-Z]\w*(?=[(<])/,
+        lookbehind: true,
+        alias: 'function',
+    },
+    'keyword': /\b(?:bitcast|break|case|const|continue|continuing|default|discard|else|enable|fallthrough|fn|for|function|if|let|loop|private|return|storage|switch|type|uniform|var|while|workgroup)\b/,
+    'function-calls': {
+        pattern: /\b[_a-z]\w*(?=\()/i,
+        alias: 'function',
+    },
+    'builtin': /\b(?:array|atomic|bool|i32|u32|f32|i64|u64|f64|vec[2-4]|mat[2-4]x[2-4]|override|ptr|sampler|sampler_comparison|staticAssert|struct|texture_1d|texture_2d|texture_2d_array|texture_3d|texture_cube|texture_cube_array|texture_multisampled_2d|texture_storage_1d|texture_storage_2d|texture_storage_2d_array|texture_storage_3d|texture_depth_2d|texture_depth_2d_array|texture_depth_cube|texture_depth_cube_array|texture_depth_multisampled_2d)\b/,
```

Comment:
Please match bulti-ins before `function-calls`.
![image](https://user-images.githubusercontent.com/20878432/168776361-40c7e27a-7107-4879-8496-d45e0e40c31c.png)

---

`struct` is a keyword, not a built-in type or function.

![image](https://user-images.githubusercontent.com/20878432/168777326-2de30955-73ea-4d0c-84e4-04d0fc1ae957.png)